src/main/java/com/epam/test_generator/services/StepSuggestionService.java

```diff
@@ -30,143 +26,92 @@
      *
-     * @param stepSuggestionId        id of step suggestion to update
+     * @param stepSuggestionId id of step suggestion to update
      * @param stepSuggestionUpdateDTO info to update
      */
-    public void updateStepSuggestion(Long stepSuggestionId,
+    public StepSuggestionDTO updateStepSuggestion(Long projectId,
+                                     Long stepSuggestionId,
                                      StepSuggestionUpdateDTO stepSuggestionUpdateDTO) {
-        StepSuggestion stepSuggestion = stepSuggestionDAO.findOne(stepSuggestionId);
+        StepSuggestion stepSuggestion = getStepSuggestion(projectId, stepSuggestionId);
         checkNotNull(stepSuggestion);
-        verifyVersion(stepSuggestionUpdateDTO.getVersion(), stepSuggestion);
-        if (stepSuggestionUpdateDTO.getContent() != null) {
-            stepSuggestion.setContent(stepSuggestionUpdateDTO.getContent());
-        }
-
-        if (stepSuggestionUpdateDTO.getType() != null) {
-            stepSuggestion.setType(stepSuggestionUpdateDTO.getType());
-        }
-
-        stepSuggestionDAO.save(stepSuggestion);
+        stepSuggestion.verifyVersion(stepSuggestionUpdateDTO.getVersion());
+        stepSuggestionTransformer.updateFromDto(stepSuggestion, stepSuggestionUpdateDTO);
+        return stepSuggestionTransformer.toDto(
+            projectStepSuggestionDAO.save(stepSuggestion));
     }
 
     /**
      * Deletes step suggestion from database by id
      *
      * @param stepSuggestionId id of step suggestion to delete
      */
-    public void removeStepSuggestion(Long stepSuggestionId) {
-        StepSuggestion stepSuggestion = stepSuggestionDAO.findOne(stepSuggestionId);
-        checkNotNull(stepSuggestion);
+    public void removeStepSuggestion(Long projectId, Long stepSuggestionId) {
+        Project project = checkNotNull(projectDAO.getOne(projectId));
 
-        stepSuggestionDAO.delete(stepSuggestionId);
-    }
+        StepSuggestion stepSuggestion =
+            getStepSuggestion(projectId, stepSuggestionId);
 
-    /**
-     * Find all steps suggestions by string ignoring case
-     *
-     * @param searchString - string for search
-     * @param limit - count of results
-     * @return list with found steps suggestions
-     */
-    public List<StepSuggestionDTO> findStepsSuggestions(String searchString, int limit) {
-        if(searchString.isEmpty()) {
-            throw new BadRequestException("Text must not be empty!");
-        }
+        project.removeStepSuggestion(stepSuggestion);
 
-        if(limit < 1){
-            throw new BadRequestException("Limit must not be less than one!");
-        }
-        Pageable numberOfReturnedResults = new PageRequest(0, limit);
-        List<StepSuggestion> foundStepsSuggestions = stepSuggestionDAO
-            .findByContentIgnoreCaseContaining(searchString, numberOfReturnedResults);
+        projectStepSuggestionDAO.delete(stepSuggestionId);
```

Comment:
`project.removeStepSuggestion(stepSuggestion)` should do the job.But after that you should save the project. No need to call `projectStepSuggestionDAO.delete(stepSuggestionId);`

Reply:
Done.